src/components/queryExpressionEditor/fieldSelect/FieldSelect.tsx

```diff
@@ -0,0 +1,108 @@
+export const OPTIONS_PER_PAGE: number = 10;
+export const FIELD_SELECT_DEFAULT_OPTION: string = 'Select field';
+
+export interface IFieldSelectOwnProps {
+    fields: IField[];
+    expressionEditorId: string;
+}
+
+export interface IFieldSelectOwnState {
+    hasMoreFieldOptions?: boolean;
+}
+
+export interface IFieldSelectDispatchProps {
+    onOptionsChanged?: (id: string, options: IDropdownOption[]) => void;
+}
+
+export interface IFieldSelectProps extends IFieldSelectOwnProps, IFieldSelectDispatchProps {}
+
+export class FieldSelect extends React.Component<IFieldSelectProps, IFieldSelectOwnState> {
+    readonly dropdownSearchId: string;
+    private allFieldOptions: IDropdownOption[];
+    private currentFieldOptions: IDropdownOption[];
+    private optionsPage: number;
+
+    constructor(props: IFieldSelectProps) {
+        super(props);
+        this.state = {hasMoreFieldOptions: true};
+        this.dropdownSearchId = `${this.props.expressionEditorId}-${FIELD_SELECT_ID}`;
+        this.allFieldOptions = null;
+        this.currentFieldOptions = [];
+        this.optionsPage = 1;
+    }
+
+    async componentWillMount() {
+        this.allFieldOptions = await this.getFieldsOptions();
+        this.updateInfinteScroll();
+    }
+
+    private getFieldsOptions(): IDropdownOption[] {
+        const fieldsDropdownOptions: IDropdownOption[] = [];
+        _.forEach(this.props.fields, (field: IField) => {
+            const dropDownOption: IDropdownOption = {value: field.name};
+            fieldsDropdownOptions.push(dropDownOption);
+        });
+        return fieldsDropdownOptions;
+    }
+
+    private updateInfinteScroll() {
+        this.updateHasMoreItems();
+        this.updateCurrentOptions();
+        this.optionsPage++;
+    }
+
+    private updateHasMoreItems() {
+        const startingIndex: number = (this.optionsPage - 1) * OPTIONS_PER_PAGE;
+        if (startingIndex > (this.allFieldOptions.length - 1)) {
+            this.setState({hasMoreFieldOptions: false});
+        }
+    }
+
+    private updateCurrentOptions() {
+        const newOptions: IDropdownOption[] = this.getNewOptions();
+        this.currentFieldOptions = this.currentFieldOptions.concat(newOptions);
+        this.props.onOptionsChanged(this.dropdownSearchId, this.currentFieldOptions);
+    }
+
+    private getNewOptions(): IDropdownOption[] {
+        const startingIndex: number = (this.optionsPage - 1) * OPTIONS_PER_PAGE;
+        const newOptions: IDropdownOption[] = this.allFieldOptions.slice(startingIndex, startingIndex + OPTIONS_PER_PAGE);
+        return newOptions;
+    }
+
+    render() {
+        return (
+            <span className={`${styles.container}`}>
+                {/*
+                // TODO QUESTION R-V : BUG?
+                // When loading more values in the infinte scroll,
+                // The selected value resets to 'Select an option'
+                // Why does the value gets reset?

```

Comment:
@sroy3 any idea on top of your head?

Reply:
Not quite sure why this would happen, but our `DropdownSearch` is very unstable and does not handle change too well. In theory this component is even deprecated. We added the infinite scroll on it because the org picker was already using the component, but you could use another dropdown with the infinite scroll that would be a lot more stable.

Reply:
the issue was disccues with Jé